packages/react-utils/src/helpers/dataLoaders.ts

```diff
@@ -73,3 +74,34 @@ export const fetchProtectedImage = async (imageURL: string) => {
 
   return null;
 };
+
+/**
+ * Higher order function that creates the FHIR Client instance and passes token
+ * to request header
+ *
+ * Usage
+ * -----------
+ * const serve = await FHIRService("fhir base url")
+ *
+ * **To make a GET request: serve.request('fhir-resource-type')
+ *
+ * **To make a POST request: serve.create('fhir-resource-payload')
+ *
+ * **To make a PUT request: serve.update('fhir-resource-payload')
+ *
+ * **To DELETE a resource: serve.delete('<fhir-resource-type>/<id>')
+ *
+ * @param {string} fhirBaseURL - FHIR base URL
+ */
+
+export const FHIRService = async (fhirBaseURL: string) => {
+  const token = await handleSessionOrTokenExpiry();
+  const serve = new Client({} as never, {
+    serverUrl: fhirBaseURL,
+    tokenResponse: {
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      access_token: token,
+    },
+  });
+  return serve;
+};
```

Comment:
qq: how to specify the url endpoint, or should that be part of fhirBaseURL

Reply:
the usage is described in [the docstring](https://github.com/opensrp/web/pull/830/files#diff-adb3bc21623dc5c91fba98dc42776d5c0dc335dab1f4fe0ef5afb50a078ee10cR82-R92)

Reply:
As described in the docstring, the resource type is the endpoint

Reply:
cool, then how would you make a `POST` request  where you have to supply both the body and endpoint

Reply:
ok starting to make sense